projects/core/src/cms/connectors/component/cms-component.connector.ts

```diff
@@ -45,14 +47,15 @@ export class CmsComponentConnector {
         );
 
         if (missingIds.length > 0) {
-          return this.adapter
-            .findComponentsByIds(missingIds, pageContext)
-            .pipe(
-              map(loadedComponents => [
-                ...configuredComponents.filter(Boolean),
-                ...loadedComponents,
-              ])
-            );
+          return (this.config.backend.occ.legacy
+            ? this.adapter.searchComponentsByIds(missingIds, pageContext)
```

Comment:
Instead of having two methods that do the same thing, I'd suggest to add a `legacy` parameter to the original method and extract the http call.

Reply:
The point of the separation is to easily remove it when we no longer need 1811. If we introduce an if statements to do one thing, we would need to refactor them in the future. Refactoring would require several if statements which is  going to make it more difficult to refactor them in the future.

Reply:
One more thing is that it helps with the backend to know the difference between the call. For example, findComponentsByIds is the same naming convention used in the backend for the GET request, and searchComponentsByIds is the same as POST in the backend.